[Feature] How to create view in FlinkGenericCatalog ?

### Search before asking

- [x] I searched in the [issues](https://github.com/apache/paimon/issues) and found nothing similar.


### Motivation

I have a Flink Generic Catalog which Hive Metastore. 

But i want to create a view. It seem prompt me 

```sql
create view paimon_gen_catalog.mydb.`my_overall_share` as 
	select * from paimon_gen_catalog.mydb.`share` where id = 1313;

```

Caused by: java.lang.RuntimeException: FlinkGenericCatalog can not create table without 'connector' key.
	at org.apache.paimon.flink.FlinkGenericCatalog.createTable(FlinkGenericCatalog.java:184) ~[paimon-flink-1.20-1.0.1.jar:1.0.1]

### Solution

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkGenericCatalog.java
+++ b/paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkGenericCatalog.java
@@ -27,6 +27,7 @@ import org.apache.flink.table.catalog.CatalogDatabase;
 import org.apache.flink.table.catalog.CatalogFunction;
 import org.apache.flink.table.catalog.CatalogPartition;
 import org.apache.flink.table.catalog.CatalogPartitionSpec;
+import org.apache.flink.table.catalog.CatalogView;
 import org.apache.flink.table.catalog.ObjectPath;
 import org.apache.flink.table.catalog.TableChange;
 import org.apache.flink.table.catalog.exceptions.CatalogException;
@@ -179,6 +180,15 @@ public class FlinkGenericCatalog extends AbstractCatalog {
     @Override
     public void createTable(ObjectPath tablePath, CatalogBaseTable table, boolean ignoreIfExists)
             throws TableAlreadyExistException, DatabaseNotExistException, CatalogException {
+        // create view
+        // TODO: By default, use the Paimon catalog to create the view. Create views with options is
+        // not supported.
+        if (table instanceof CatalogView) {
+            paimon.createTable(tablePath, table, ignoreIfExists);
+            return;
+        }
+
+        // create table
         String connector = table.getOptions().get(CONNECTOR.key());
         if (connector == null) {
             throw new RuntimeException(

--- a/paimon-hive/paimon-hive-connector-common/src/test/java/org/apache/paimon/hive/FlinkGenericCatalogITCase.java
+++ b/paimon-hive/paimon-hive-connector-common/src/test/java/org/apache/paimon/hive/FlinkGenericCatalogITCase.java
@@ -217,4 +217,15 @@ public class FlinkGenericCatalogITCase extends AbstractTestBaseJUnit4 {
         List<Row> result = sql("SELECT tag_name FROM paimon_t$tags");
         assertThat(result).contains(Row.of("tag_1"));
     }
+
+    @Test
+    public void testCreateView() {
+        sql("CREATE TABLE paimon_t ( " + "f0 INT, " + "f1 INT " + ") WITH ('connector'='paimon')");
+        sql("INSERT INTO paimon_t VALUES (1, 1), (2, 2)");
+        assertThat(sql("SELECT * FROM paimon_t"))
+                .containsExactlyInAnyOrder(Row.of(1, 1), Row.of(2, 2));
+        sql("CREATE VIEW paimon_t_view AS SELECT * FROM paimon_t WHERE f0=1");
+
+        assertThat(sql("SELECT * FROM paimon_t_view")).containsExactlyInAnyOrder(Row.of(1, 1));
+    }
 }

